Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -694,6 +694,100 @@ extension PostgresConnection {
     }
 }
 
+// MARK: Copy from
+
+fileprivate extension EventLoop {
+    /// If we are on the given event loop, execute `task` immediately. Otherwise schedule it for execution.
+    func executeImmediatelyOrSchedule(_ task: @Sendable @escaping () -> Void) {
+        if inEventLoop {
+            return task()
+        }
+        return execute(task)
+    }
+}
+
+/// A handle to send 
+public struct PostgresCopyFromWriter: Sendable {
+    private let channelHandler: NIOLoopBound<PostgresChannelHandler>
+    private let context: NIOLoopBound<ChannelHandlerContext>
```

Comment:
use the `handlerContext: ChannelHandlerContext?` with `!` in `PostgresChannelHandler` instead.

Reply:
You don't need to pass the context to `sendCopyDone` or `copyData`. Instead use the context that already exists in the `PostgresChannelHandler.handlerContext`.

Reply:
Oops, not sure how I missed that.